### PR TITLE
Internal: Dynamic Transformer

### PR DESCRIPTION
**Pull request by Nevoss**

### CODE REVIEW

File: tests/phpunit/elementor/modules/atomic-widgets/test-atomic-widget-base.php [hunk 1]
```diff
@@ -23,72 +25,153 @@ class Test_Atomic_Widget_Base extends Elementor_Test_Base {
...
 	 * @dataProvider get_atomic_settings_data_provider
 	 */
-	public function test_get_atomic_settings( $prop_types, $settings, $result ) {
+	public function test_get_atomic_settings( $args, $arrange_cb = null ) {
 		// Arrange.
+		$cleanup = null;
+
+		if ( $arrange_cb ) {
+			$cleanup = $arrange_cb();
+		}
```
Comment (StyleShit) [inline tests/phpunit/elementor/modules/atomic-widgets/test-atomic-widget-base.php:34]:
```suggestion
		$cleanup = $arrange_cb ? $arrange_cb() : fn() => {};
```

maybe?

(and then remove the condition at the end)
(but not critical)

File: tests/phpunit/elementor/modules/atomic-widgets/test-atomic-widget-base.php [hunk 2]
```diff
@@ -23,72 +25,153 @@ class Test_Atomic_Widget_Base extends Elementor_Test_Base {
...
+					],
+				],
+				'arrange_cb' => function() {
+					Plugin::$instance->dynamic_tags->register(
+						new class() extends Tag {
+							public function get_name() { return 'dynamic-tag'; }
+							public function get_group() { return 'dynamic'; }
+							public function get_categories() { return [ 'text' ]; }
+							public function get_title() { return 'Dynamic Tag'; }
+							public function get_content( array $options = [] ) {
```
Comment (StyleShit) [inline tests/phpunit/elementor/modules/atomic-widgets/test-atomic-widget-base.php:164]: I wonder if we should just create a standalone test
it's kinda complicated
WDYT?

**Review by StyleShit — CHANGES_REQUESTED** [review]
image

File: tests/phpunit/elementor/modules/atomic-widgets/test-atomic-widget-base.php (hunk 1 above)
Comment (Nevoss) [reply tests/phpunit/elementor/modules/atomic-widgets/test-atomic-widget-base.php:34]: Beautiful 🌷 Done.

**Review by StyleShit — APPROVED** [review]

**Pull request merged**